src/mango/src/mango_cursor_view.erl

```diff
@@ -248,6 +266,19 @@ view_cb({row, Row}, #mrargs{extra = Options} = Acc) ->
         key = couch_util:get_value(key, Row),
         doc = couch_util:get_value(doc, Row)
     },
+    % This supports receiving our "arguments" either as just the `selector`
+    % or in the new record in `callback_args`. This is to support mid-upgrade
+    % clusters where the non-upgraded coordinator nodes will send the older style.
+    % TODO remove this in a couple of couchdb versions.
```

Comment:
might be worth codifying this a little. I know we have other places, but maybe e can start here:

maybe we can make a comment like `% x-couch-remove: 4.0.0` that we can then grep for prior to releasing 4.0.0.

The reasoning here that we can say: upgrade everything to the latest 3.x version and THEN go to 4.0.0 will be smooth, rather than supporting any 3.x-> 4.0.0

Reply:
This seems reasonable. My question is: does this removal need to wait for 4.x, given it's not a breaking change, or does it need t wait a couple of minor - or patch - releases?

I don't know how many people do rolling upgrades outside of Cloudant, as opposed to just stopping the world for a few minutes.

Reply:
I think enough folks do this and not necessarily within a small round of point releases that would warrant keeping it up until 4.0.0

Reply:
but I’m fine with making this change in a follow-up PR where we go through all our rolling update compat code and merge this as-is #ScopeCreep

Reply:
Okay. Sounds good.